modules/pricer/src/main/java/com/opengamma/strata/pricer/rate/bond/DiscountingFixedCouponBondTradePricer.java

```diff
@@ -0,0 +1,1076 @@
+    LocalDate settlementDate = trade.getTradeInfo().getSettlementDate().get();
+    if (expanded.getPeriodicPayments().get(0).getStartDate().isAfter(settlementDate)) {
+      return 0d;
+    }
+    int couponIndex = couponIndex(expanded.getPeriodicPayments(), settlementDate);
+    double factorSpot = accruedInterest(trade) / product.getFixedRate() / product.getNotional();
+    double factorPeriod = expanded.getPeriodicPayments().get(couponIndex).getYearFraction();
+    return (factorPeriod - factorSpot) / factorPeriod;
+  }
+
+  private int couponIndex(Schedule schedule, LocalDate date) {
+    int nbCoupon = schedule.getPeriods().size();
+    int couponIndex = 0;
+    for (int loopcpn = 0; loopcpn < nbCoupon; ++loopcpn) {
+      if (schedule.getPeriods().get(loopcpn).getEndDate().isAfter(date)) {
+        couponIndex = loopcpn;
+        break;
+      }
+    }
+    return couponIndex;
+  }
+
+  private int couponIndex(ImmutableList<FixedCouponBondPaymentPeriod> list, LocalDate date) {
+    int nbCoupon = list.size();
+    int couponIndex = 0;
+    for (int loopcpn = 0; loopcpn < nbCoupon; ++loopcpn) {
+      if (list.get(loopcpn).getEndDate().isAfter(date)) {
+        couponIndex = loopcpn;
+        break;
+      }
+    }
+    return couponIndex;
+  }
+
+  //-------------------------------------------------------------------------
+  private CurrencyAmount presentValueCoupon(
+      ExpandedFixedCouponBond product,
+      IssuerCurveDiscountFactors discountFactors,
+      LocalDate settlementDate,
+      boolean exCoupon) {
+
+    double total = 0d;
+    for (FixedCouponBondPaymentPeriod period : product.getPeriodicPayments()) {
+      if ((exCoupon && !settlementDate.isAfter(period.getDetachmentDate())) ||
+          (!exCoupon && period.getPaymentDate().isAfter(settlementDate))) {
+        total += periodPricer.presentValue(period, discountFactors);
+      }
+    }
+    return CurrencyAmount.of(product.getCurrency(), total);
+  }
+
+  private CurrencyAmount presentValueCouponFromZSpread(
+      ExpandedFixedCouponBond product,
+      IssuerCurveDiscountFactors discountFactors,
+      double zSpread,
+      boolean periodic,
+      int periodsPerYear,
+      LocalDate settlementDate,
+      boolean exCoupon) {
+
+    double total = 0d;
+    for (FixedCouponBondPaymentPeriod period : product.getPeriodicPayments()) {
+      if ((exCoupon && !settlementDate.isAfter(period.getDetachmentDate())) ||
+          (!exCoupon && period.getPaymentDate().isAfter(settlementDate))) {
+        total += periodPricer.presentValueWithSpread(period, discountFactors, zSpread, periodic, periodsPerYear);
+      }
+    }
+    return CurrencyAmount.of(product.getCurrency(), total);
+  }
+
+  private CurrencyAmount presentValueNominal(
+      ExpandedFixedCouponBond product,
+      IssuerCurveDiscountFactors discountFactors,
+      LocalDate settlementDate) {
+
+    Payment nominal = product.getNominalPayment();
+    if (!settlementDate.isAfter(nominal.getDate())) {
```

Comment:
We may want to add that check in the trade builder, but I don't think this check is required at the pricing level.